src/agent/src/initdata.rs

```diff
@@ -0,0 +1,636 @@
+        }
+        if attempts >= timeout_secs {
+            break;
+        }
+        attempts += 1;
+        info!(
+            logger,
+            "waiting for {} to exist (attempts={})", path, attempts
+        );
+    }
+
+    Err(anyhow!("wait for {} to exist timeout.", path))
+}
+
+pub async fn launch_process(
+    logger: &Logger,
+    path: &str,
+    args: Vec<String>,
+    unix_socket_path: &str,
+    timeout_secs: i32,
+) -> Result<()> {
+    if !Path::new(path).exists() {
+        bail!("path {} does not exist.", path);
+    }
+    if !unix_socket_path.is_empty() && Path::new(unix_socket_path).exists() {
+        fs::remove_file(unix_socket_path).await?;
+    }
+    Command::new(path).args(args).spawn()?;
+    if !unix_socket_path.is_empty() && timeout_secs > 0 {
+        wait_for_path_to_exist(logger, unix_socket_path, timeout_secs).await?;
+    }
+
+    Ok(())
+}
+
+async fn launch_confidential_data_hub(cdh_config: Option<String>) -> Result<()> {
+    if Path::new(CDH_PATH).exists() {
+        let args = match cdh_config {
+            Some(config) => {
+                debug!(
+                    sl!(),
+                    "Launch Confidential Data Hub with config delivered via Initdata"
+                );
+                tokio::fs::write(CDH_CONFIG_PATH, config.as_bytes()).await?;
+                vec!["-c".into(), CDH_CONFIG_PATH.into()]
+            }
+            None => {
+                debug!(sl!(), "Launch Confidential Data Hub with default config");
+                vec![]
+            }
+        };
+
+        let ocicrypt_config = serde_json::json!({
+            "key-providers": {
+                "attestation-agent":{
+                    "ttrpc":CDH_SOCKET_URI
+                }
+            }
+        });
+
+        fs::write(OCICRYPT_CONFIG_PATH, ocicrypt_config.to_string().as_bytes()).await?;
+        env::set_var("OCICRYPT_KEYPROVIDER_CONFIG", OCICRYPT_CONFIG_PATH);
+
+        launch_process(
+            &sl!(),
+            CDH_PATH,
+            args,
+            CDH_SOCKET,
+            DEFAULT_LAUNCH_PROCESS_TIMEOUT,
+        )
+        .await?;
+        debug!(
+            sl!(),
+            "Confidential Data Hub has been launched successfully."
+        );
+    }
+
+    Ok(())
+}
+
+async fn launch_attestation_agent(aa_config: Option<String>, initdata: String) -> Result<()> {
+    if Path::new(AA_PATH).exists() {
+        let mut args = match aa_config {
+            Some(config) => {
+                debug!(
+                    sl!(),
+                    "Launch Attestation Agent with config delivered via Initdata"
+                );
+                tokio::fs::write(AA_CONFIG_PATH, config.as_bytes()).await?;
+                vec!["-c".into(), AA_CONFIG_PATH.into()]
+            }
+            None => {
+                debug!(sl!(), "Launch Attestation Agent with default config");
+                vec![]
+            }
+        };
+
+        args.push("--initdata".into());
```

Comment:
can we specify the initdata digest as file w/ raw bytes? (e.g `/run/initdata.digest`).

That would make it easier if AA is launched via systemd. otherwise, the logic of picking the hash alg and calculating the digest has to be reimplemented and performed elsewhere.